src/sentry/integrations/vercel/integration.py

```
@@ -248,55 +248,56 @@ def update_organization_config(self, data):
+            }
 
-            secrets.append("")
-            for secret, env_var in zip(secrets, env_var_names):
-                self.create_env_var(vercel_client, vercel_project_id, env_var, secret)
+            for env_var, details in env_var_map.items():
+                if details["type"] == "secret":
+                    secret_name = env_var + "_%s" % uuid
+                    details["value"] = vercel_client.create_secret(secret_name, details["value"])
+                self.create_env_var(
+                    vercel_client, vercel_project_id, env_var, details["value"], details["type"]
+                )
 
         config.update(data)
         self.org_integration.update(config=config)
 
-    def env_var_already_exists(self, client, vercel_project_id, name):
-        return any(
-            [
-                env_var
-                for env_var in client.get_env_vars(vercel_project_id)
-                if env_var["key"] == name
-            ]
-        )
+    def create_env_var(self, client, vercel_project_id, key, value, type):
+        data = {
+            "key": key,
+            "value": value,
+            "target": ["production"],
+            "type": type,
+        }
+        try:
+            return client.create_env_variable(vercel_project_id, key, value, data)
+        except ApiError as e:
+            if e.json and e.json.get("error", {}).get("code"):
+                if e.json["error"]["code"] == "ENV_ALREADY_EXISTS":
+                    return self.update_env_variable(client, vercel_project_id, key, value, data)
+            raise
+
+    def update_env_variable(self, client, vercel_project_id, key, value, data):
+        env_var_id = [
+            env_var["id"]
+            for env_var in client.get_env_vars(vercel_project_id)["envs"]
+            if env_var["key"] == key
+        ]
+        if env_var_id:
+            return client.update_env_variable(vercel_project_id, env_var_id[0], data)
```

@ceorourke this is where we need to throw an error if we don't find `env_var_id`

---

nvm, i see it below